### PR TITLE
MB-12967: counseling report page

### PR DESCRIPTION
**Pull request by leia-sefkin**
## Summary

This PR adds basic counseling report functionality including creating a new counseling report, the form, save as draft, and canceling a report. It includes both the front and backend functionality needed to do this. Also included is quite a bit of refactoring and cleanup around the balance of using common components for shipment vs counseling report types.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

## Verification Steps for Reviewers

- [ ] New counseling reports can be created with the 'Create report' button on the evaluation reports page
- [ ] Counseling reports can be canceled (soft deleted) using the cancel button
- [ ] Counseling reports can be saved as draft with the 'save draft' button
- [ ] The counseling report does not include the shipment form specific fields (destination radio options and observed delivery/pickup dates)
- [ ] When the violations radio is 'Yes' the submit button changes to 'Next: select violations' and takes the user to the violations selection page (just a stub page)
- [ ] Refactors to commonize Shipment vs Counseling report make sense
- [ ] Backend endpoints are appropriate/reasonable
- [ ] Tests and story book stories were added where reasonable

https://user-images.githubusercontent.com/62263329/184971906-cb896c05-eeb6-4069-a387-06a9232a4ba9.mov

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1 +1,2 @@
 golang 1.18.4
+nodejs 16.15.0
```
Comment (PDickeyTruss) [inline .tool-versions:2]: I admittedly do not know what this file is about or does... happy to revert this change if others opinions or knowledge?

**Review by Ryan-Koch — APPROVED** [review]

File: .tool-versions (hunk 1 above)
Comment (Ryan-Koch) [reply .tool-versions:2]: Interestingly on `master` we don't have an entry for nodejs. On the go side I think it's for folks using `asdf` though I suppose there's likely tooling on the nodejs side that might benefit. The safe move might be to go ahead and yank it for now.

Comment (PDickeyTruss) [reply .tool-versions]: Agree that removing likely safest,  pulled that out ✅

**Review by cosburn — APPROVED** [review]
Looks good to me!
